dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -40,43 +39,100 @@ export function Table(options: TableData) {
   };
 
   const columns = [
+    {
+      field: 'type',
+      name: '',
+      width: '3%',
+      render: (item) => {
+        if (item == 'Visualization') {
+          return (
+            <div>
+              <EuiLoadingChart size="m" />
+            </div>
+          );
+        } else {
+          return (
+            <div>
+              <EuiIcon type="discoverApp" size="m" />
+            </div>
+          );
+        }
+      },
+    },
     {
       field: 'data',
       name: 'Name',
-      render: (item)=>{return <EuiLink onClick={() =>
-      {options.savedQuerySearch(item.query, [item.date_start, item.date_end], item.timestamp, item.fields)}}>
-        {item.name}
-      </EuiLink>},
+      align: 'left',
+      render: (item) => {
+        return (
+          <EuiLink
+            onClick={() => {
+              options.savedQuerySearch(
+                item.query,
+                [item.date_start, item.date_end],
+                item.timestamp,
+                item.fields
+              );
+            }}
+          >
+            {item.name}
+          </EuiLink>
+        );
+      },
     },
     {
-      field: 'description',
-      name: 'Description',
+      field: 'type',
+      name: 'Type',
+      align: 'right',
     },
   ];
 
-
-
+  let objectName = '';
   const queries = options.savedHistory.map((h) => {
     const savedObject = h.hasOwnProperty('savedVisualization')
```

Comment:
if you need to check if it is savedVisualization more than once, you could do just const isSavedVisualization = h.hasOwnProperty('savedVisualization'), and then use isSavedVisualization in the following code.

Reply:
I'm checking if its savedVisualization or not for each object, so it may differ for each entry of data. as i need to display the type of the query as a column, i need to populate it

Reply:
I mean in your each loop.

Reply:
Got it